thingif/src/main/java/com/kii/thingif/ThingIFAPI.java

```diff
@@ -892,42 +871,46 @@ public Trigger postNewTrigger(
             @Nullable TriggerOptions options)
         throws ThingIFException
     {
-        // TODO: implement me.
-        return null;
+        return postNewTriggerWithForm(form, predicate, options);
     }
 
-    private Trigger postNewTriggerWithCommands(
-            String schemaName,
-            int schemaVersion,
-            List<Action> actions,
-            Predicate predicate,
-            Target commandTarget) throws ThingIFException {
+    private Trigger postNewTriggerWithForm(
+            @NonNull TriggeredCommandForm form,
+            @NonNull Predicate predicate,
+            @Nullable TriggerOptions options)
+        throws ThingIFException
+    {
         if (this.target == null) {
             throw new IllegalStateException("Can not perform this action before onboarding");
         }
-        if (TextUtils.isEmpty(schemaName)) {
-            throw new IllegalArgumentException("schemaName is null or empty");
-        }
-        if (actions == null || actions.size() == 0) {
-            throw new IllegalArgumentException("actions is null or empty");
+        if (form == null) {
+            throw new IllegalArgumentException("form is null.");
         }
         if (predicate == null) {
```

Comment:
@rizakii 

I discussed this issue with @satoshikumano san.
I think our issue is followings:

1. Should we check null for @NonNull argument.
1. Should we add annotations for private method.

For 1, kumano san said that we follow other implemented API's way. So I think We should check null for @NonNull argument.

For 2, kumano san said that adding annotations to private method is better manner, We can add annotations to private method.

So I think checking null for @NonNull argument and adding annotations to private method is OK. What do you think?